### DISCUSSION TITLE
How to know if a Queue was deleted while the consumer is connected to the server?

### DISCUSSION THREAD

**Discussion by wilrobg** [rabbitmq/rabbitmq-dotnet-client#1133] [General]
As the title says,
I want to know if there is a way to let know the consumer if the Queue was deleted while a consumer is using this queue.

I make a test where I delete the queue and the consumer didn't react to this. I was expecting that the consumer will shutdown or disconnected from the server, also was expecting that maybe the consumer will create again the queue.

Has someone experimented with this behavior? Also is it make sense for what I'm looking for?

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment]
Using [Consumer Cancelation Notifications](https://www.rabbitmq.com/consumer-cancel.html) which this client should support.

**Reply by wilrobg** [reply]
Thanks Michale for your answer!

To be more precise, the queue is removed directly from the server.

This will trigger somehow the deleted notification?
By the way I'm using C#

```
_consumer.Received += async (model, ea) =>
  {
       _logger.LogWarning("Received");
  };

  _consumer.OnCancel(_consumer.ConsumerTags);

  _consumer.Shutdown += async (model, ea) =>
  {
      _logger.LogWarning("Server is down");
  };
```

**Comment by wilrobg** [comment]
Sorry I didn't realize that EventingBasicConsumer has the event ConsumerCancelled to handle cancelled notification by the server!
Thanks for your help!
